src/example/prompt.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Injectable()
+export class PullRequestReviewService {
+  constructor(private readonly httpService: HttpService) {}
+
+  async getPullRequestDetails(repo: string, prId: number): Promise<any> {
+    const url = `https://api.github.com/repos/${repo}/pulls/${prId}`;
+    return this.httpService.get(url).pipe(map(response => response.data)).toPromise();
+  }
```

Comment:
PullProofAI

The use of `toPromise()` is deprecated in RxJS 7 and will be removed in future versions. Consider using `lastValueFrom` or `firstValueFrom` instead for converting an Observable to a Promise.

```diff
-    return this.httpService.get(url).pipe(map(response => response.data)).toPromise();
+    return lastValueFrom(this.httpService.get(url).pipe(map(response => response.data)));
```

Ensure you import `lastValueFrom` from `rxjs`.



<!-- This is an auto-generated comment by PullProofAI -->